### ISSUE TITLE
rasterimage return partial image

### ISSUE THREAD

**Issue by meravwebs** [meyfa/php-svg#131]
i'm using this code:
$image = SVG::fromString($svg);
$doc = $image->getDocument();

$w = $doc->getWidth();
$h = $doc->getHeight();

$rasterImage = $image->toRasterImage($w, $h);

imagepng($rasterImage, $png_path);

it creates an image, but only partial.

the svg i'm using have image element in it and the image created is only part of the image element of the svg instead of the entire svg.

**Comment by jcogs-design** [comment]
I have this issue too... a composite SVG that includes a raster image, some gradients, shapes and a text path loads correctly (all elements are present in the `SVG\SVG` type object created by the `SVG::fromString` method, but when a raster version of the image is created, while the dimensions are correct for the whole svg source image, the only content is the embedded raster image.

Using php 7.4... 

Any guidance / hints on how to resolve would be very helpful!

**Comment by meyfa** [comment]
The way to resolve this is to implement the required functionality in this library. At the moment, rendering of gradients is not supported. Text rendering is severely limited as well. In the past few weeks I have spent some time improving rasterization, such as by introducing `transform` support, improved path and polygon filling, style parsing and a few other details. If your SVGs make use of these features, you will see improvements with the upcoming v0.12.0 release. There is still a lot missing though. Feel free to contribute code for some of the missing features :) Otherwise, if you just need a robust solution quickly, I hear some people have had more luck with Gmagick or Imagick for converting SVG to raster formats.

**Comment by meyfa** [comment]
> Probably would be helpful if the readme gave some better details about what the library is able to do

Yes, I absolutely agree! Will add "summary of the current feature set" to the to-do-list in #125.

> the mistaken understanding that rendering of text or shapes was supported

Actually, rendering shapes _should_ be supported. All of the basic geometric shapes (ellipses, rectangles, polygons...) as well as paths support `fill` and `stroke` rendering. Can you provide a short sample SVG showing the stuff that doesn't work?

**Comment by jcogs-design** [comment]
Ace - look forward to seeing the summary when it is avail.

Another thing that would be useful perhaps would be your thoughts about the priorities for development.

The test image I used is attached - it is something created in Pixelmator Pro, and when rendered using the library only the embedded image is shown - the text and star shapes it includes do not get rendered.


![svg_test](https://user-images.githubusercontent.com/13821249/156456145-85fe2b20-1bee-4f03-a601-a33aec9be850.svg)
